samples/excel/50-workbook/create-workbook.yaml

```diff
@@ -0,0 +1,86 @@
+order: 2
+id: excel-workbook-create-workbook
+name: Create workbook
+description: Sample for Excel
+author: OfficeDev
+host: EXCEL
+api_set:
+    ExcelApi: 1.8
+script:
+    content: |
+        $("#file").change(() => tryCatch(createWorkbook));
+
+        async function createWorkbook() {
+            const myFile = document.getElementById("file");
+            const reader = new FileReader();
+
+            reader.onload = ((theFile) => {
+                return ((event) => {
+                    Excel.run(context => {
+                        // strip off the metadata before the base64-encoded string
+                        const startIndex = event.target.result.indexOf("base64,");
+                        const mybase64 = event.target.result.substr(startIndex + 7);
+
+                        Excel.createWorkbook(mybase64);
+                        return context.sync();
+                    });
+                });
+            })(myFile.files[0]);
```

Comment:
I get an error from myFile.files[0] saying that HTMLElement does not have a property named files.

Reply:
I do too, but that's how that API works ([SO example here](https://stackoverflow.com/questions/8645369/how-do-i-get-the-file-content-from-a-form)). Am I supposed to import something?

Reply:
@AlexJerabek  You have to do some casting. The same problem and solution applies to EventTarget.result. Here's what the code should look like:

```
const myFile = <HTMLInputElement>document.getElementById("file");
    const reader = new FileReader();
    reader.onload = ((event) => {
        Excel.run(context => {
            // strip off the metadata before the base64-encoded string 
            const startIndex = (<FileReader>event.target).result.indexOf("base64,");
            const mybase64 = (<FileReader>event.target).result.substr(startIndex + 7);
            Excel.createWorkbook(mybase64);
            return context.sync();
        });
```